opm/autodiff/BlackoilModelBase_impl.hpp

```diff
@@ -757,17 +780,16 @@ typedef Eigen::Array<double,
         asImpl().assembleMassBalanceEq(state);
 
         // -------- Well equations ----------
-        IterationReport iter_report = {false, false, 0, 0};
         if ( ! wellsActive() ) {
-            return iter_report;
+            return;
         }
 
         std::vector<ADB> mob_perfcells;
         std::vector<ADB> b_perfcells;
         asImpl().wellModel().extractWellPerfProperties(state, sd_.rq, mob_perfcells, b_perfcells);
         if (param_.solve_welleq_initially_ && initial_assembly) {
             // solve the well equations as a pre-processing step
-            iter_report = asImpl().solveWellEq(mob_perfcells, b_perfcells, reservoir_state, state, well_state);
+            asImpl().solveWellEq(mob_perfcells, b_perfcells, reservoir_state, state, well_state);
```

Comment:
This is a significant change, if the pre-well-solve goes bad we should not lose that info.